### PR TITLE
Update authentication due to a thread pool bean added

### PR DESCRIPTION
**Pull request by moarychan**
Fixes https://github.com/Azure/azure-sdk-for-java/issues/28525

Add a note for the thread pool bean created by Spring Cloud Azure.

### CODE REVIEW

File: docs/src/main/asciidoc/authentication.adoc [hunk 1]
```diff
@@ -36,6 +36,7 @@ image::https://user-images.githubusercontent.com/13167207/143148654-f3a37180-85e
 
 TIP: Please refer to link:index.html#authorize-access-with-azure-active-directory[Authorize access with Azure AD] to make sure the security principal has been granted the sufficient permission to access the Azure resource.
 
+NOTE: As of Spring Cloud Azure AutoConfigure 4.1.0, a `ThreadPoolTaskExecutor` bean named **springCloudAzureCredentialTaskExecutor** will be automatically registered by default, which will manage all threads related to Azure credential. Each thread name is prefixed with **az-identity**, which is independent of the bean `Executor` provided by Spring Boot by default.
```
Comment (saragluna) [inline docs/src/main/asciidoc/authentication.adoc:39]: which will manage all threads created by Azure Identity.

Comment (saragluna) [reply docs/src/main/asciidoc/authentication.adoc:39]: **az-identity-**

Comment (saragluna) [reply docs/src/main/asciidoc/authentication.adoc:39]: Each thread name managed by this thread pool is prefixed with **az-identity-**. This `ThreadPoolTaskExecutor` bean is independent of the `Executor` bean provided by Spring Boot.

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]

**Review by hui1110 — APPROVED** [review]

**Comment by hui1110** [comment]
https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/3035

**Comment by chenrujun** [comment]
Closing this PR.
@hui1110 , please update MS docs first, then sync to this repo.

**Pull request closed without merging**